src/components/Checkbox/hooks/useFirefoxShiftLabelSupport.jsx

```diff
@@ -0,0 +1,23 @@
+import { useCallback, useMemo } from "react";
+import { detect } from "detect-browser";
+
+const BROWSER_NAME = "firefox";
+
+export function useFirefoxShiftLabelSupport({ inputRef }) {
+  const isFirefox = useMemo(() => {
+    const browser = detect();
+    console.log(browser);
+    return browser && browser.name && browser.name === BROWSER_NAME;
+  }, []);
+  const onClickCapture = useCallback(
+    e => {
+      if (e.shiftKey && isFirefox && inputRef?.current?.click) {
+        e.preventDefault();
```

Comment:
stop prorogation does not block default checkbox behavior, only prevent default 
this is what I found about the subject: https://stackoverflow.com/questions/67612404/why-does-event-stoppropagation-not-prevent-a-label-from-checking-its-checkbo
in addition, i am not sure we want to block other on click events that we will want to add in the future (we will want the same behavior for firefox as other browsers no?)
so the question is if i should add here call for stop prorogation in addition to the prevent default call

Reply:
cool!